recirq/qcqmc/hamiltonian.py

```diff
@@ -229,6 +221,107 @@ def get_restricted_fqe_hamiltonian(self) -> RestrictedHamiltonian:
             h1e=self.one_body_integrals, h2e=self.two_body_integrals_pqrs
         )
 
+    @classmethod
+    def build_hamiltonian_from_file(
+        cls,
+        params: HamiltonianFileParams,
+    ) -> 'HamiltonianData':
+        """Function for loading a Hamiltonian from a file.
+
+        Args:
+            params: parameters defining the hamiltonian.
+
+        Returns:
+            properly constructed HamiltonianFileParams object.
+        """
+        filepath = data.get_integrals_path(name=params.integral_key)
+        e_core, one_body_integrals, two_body_integrals_psqr, e_fci = load_integrals(
+            filepath=filepath
+        )
+
+        n_orb = params.n_orb
+
+        if params.do_eri_restore:
+            two_body_integrals_psqr = ao2mo.restore(1, two_body_integrals_psqr, n_orb)
+            # This step may be necessary depending on the format in which the integrals were stored.
+
+        # We have to reshape to a four index tensor and reorder the indices from
+        # chemist's ordering to physicist's.
+        two_body_integrals_psqr = two_body_integrals_psqr.reshape(
+            (n_orb, n_orb, n_orb, n_orb)
+        )
+        two_body_integrals_pqrs = np.einsum("psqr->pqrs", two_body_integrals_psqr)
+
+        fqe_ham = integrals_to_fqe_restricted(
+            h1e=one_body_integrals, h2e=two_body_integrals_pqrs
+        )
+
+        initial_wf = fqe.Wavefunction([[params.n_elec, 0, params.n_orb]])
+        initial_wf.set_wfn(strategy="hartree-fock")
+        e_hf = _cast_to_float(initial_wf.expectationValue(fqe_ham) + e_core)
+
+        return HamiltonianData(
+            params=params,
+            e_core=e_core,
+            one_body_integrals=one_body_integrals,
+            two_body_integrals_pqrs=two_body_integrals_pqrs,
+            e_hf=e_hf,
+            e_fci=e_fci,
+        )
+
+    @classmethod
+    def build_hamiltonian_from_pyscf(
```

Comment:
nit: can probably simplify the method name since you know it will be prepended with `HamiltonianData.`

Reply:
do you mean remove the "_from"?